include/visitors/interpreter.h

```diff
@@ -417,6 +423,41 @@ class Interpreter : public Visitor
         }
     }
 
+    void visit_for_stmt(ForStmt *for_stmt)
+    {
+        if (for_stmt->initializer.has_value())
+        {
+            for_stmt->initializer.value()->accept(this);
```

Comment:
I think this should happen in it's own environment or it should be cleaned up after use